### DISCUSSION TITLE
additional game executables option

### DISCUSSION THREAD

**Discussion by BachLai** [kel-z/HSR-Scanner#91] [Ideas]
Is it possible to include an option that lets you update the scanner with additional executables? This is just so that you can include other 3rd party windows (mainly cloud platforms like GeForce Now that has official HSR support) that are not StarRail or YuanShen to also run the OCR, which I don't think should have any issue as long as it is consistent image streaming.
Example of concept from InventoryKamera (for Genshin):
![image](https://github.com/kel-z/HSR-Scanner/assets/68167041/7ab5a190-a5d2-4ee9-9d01-450fc23e136c)

**Comment by kel-z** [comment]
Currently all the possible window names are hardcoded into a list and the scanner will look for any window name that matches. If there's a free trial for GeForce NOW, I'll add that name to the list and test for compatibility. Ideally this wouldn't require any other code changes.

**Reply by BachLai** [reply]
Thanks for the reply. GeForce NOW does indeed have a free tier for users that you can easily access with their desktop app (https://www.nvidia.com/en-us/geforce-now/download/), the window name should be just GeForceNOW.exe.

**Comment by kel-z** [comment]
After some initial testing, light cones and relics scan fine, but it fails when scanning characters because it's unable to drag-scroll to the next page of characters. And with no other shortcuts (hotkeys) to navigate between characters, there doesn't seem to be an easy solution for this. 

For the time being, I can make it so that scanning light cones and relics will work with GeForce NOW, but character scanning will be disable for it in the meantime.

**Reply by kel-z** [reply]
After some more testing, looks like there's some predictive cursor movement happening that's causing the mouse cursor to click beyond the intended target, which is problematic. It might be too buggy for now to include GeForce NOW as an executable option.
